### PR TITLE
chore: use name input variable for naming resources

### PR DESCRIPTION
**Pull request by gjonathanhong**

### CODE REVIEW

**Review by bradegler — APPROVED** [review]

File: terraform/github-wif.tf [hunk 1]
```diff
@@ -25,9 +25,9 @@ locals {
 resource "google_iam_workload_identity_pool" "default" {
   project = var.project_id
 
-  workload_identity_pool_id = "github-token-minter"          # 32 characters
-  display_name              = "GitHub Token Minter WIF pool" # 32 characters
-  description               = "GitHub Token Minter OIDC identity pool"
+  workload_identity_pool_id = var.name               # 32 characters
```
Comment (verbanicm) [inline terraform/github-wif.tf:28]: Are we/should we do any validation on the name variable? e.g. only alphanumeric and hyphens?

Comment (gjonathanhong) [reply terraform/github-wif.tf:28]: Yes, the name variable has existing validation here https://github.com/abcxyz/github-token-minter/blob/main/terraform/variables.tf#L25

Comment (gjonathanhong) [reply terraform/github-wif.tf:28]: For the `custom_name` input variable of the templates, I think we should add validation for the max length of the string but using the CLI to perform validations

Comment (verbanicm) [reply terraform/github-wif.tf:28]: That makes sense.

**Review by verbanicm — APPROVED** [review]

**Pull request merged**